app/controllers/admin/api/api_docs_services_controller.rb

```diff
@@ -132,8 +133,20 @@ def api_docs_params(*extra_params)
     params.require(:api_docs_service).permit(*permit_params)
   end
 
+  def api_docs_services
+    collection = current_account.api_docs_services.accessible
+    return collection unless current_user
+    collection.permitted_for(current_user)
+  end
+
   def find_api_docs_service
-    @api_docs_service = current_account.api_docs_services.find(params[:id])
+    @api_docs_service = api_docs_services.find(params[:id])
   end
 
+  def find_service
+    service_id = api_docs_params[:service_id]
+    service_id.blank? || current_user.blank? || current_user.accessible_services.find(service_id)
```

Comment:
Question 🤔  : should these endpoints work with `provider_key` ? If so, it logs in with `current_account` and without a `current_user`

Reply:
Ok, I see, the code keeps searching by `current_account` in the methods, and here it only searches the `service_id` if it is authenticated by access token 👍 
That means that indeed it should work for `provider_key` 👍

Reply:
It should continue to work with `provider_key`. And, as you can see, the method does NOT return the found service. It is a "find_service" method that behaves more like an "authorize_service" method. The reason why I didn't want to call it `authorize_service` is because we usually use `authorize_*` when relying on CanCan for the implementation of authorization (i.e. `authorize! ...`)

Reply:
Makes sense 👍